include/zenoh_commons.h

```diff
@@ -484,6 +484,25 @@ typedef struct z_get_options_t {
   enum z_query_target_t target;
   struct z_query_consolidation_t consolidation;
 } z_get_options_t;
+/**
+ * A reference-type hello message returned by a zenoh entity to a scout message sent with `z_scout`.
+ *
+ * Members:
+ *   unsigned int whatami: The kind of zenoh entity.
+ *   z_owned_bytes_t pid: The peer id of the scouted entity (empty if absent).
+ *   z_owned_str_array_t locators: The locators of the scouted entity.
+ *
+ * Like all `z_owned_X_t`, an instance will be destroyed by any function which takes a mutable pointer to said instance, as this implies the instance's inners were moved.
+ * To make this fact more obvious when reading your code, consider using `z_move(val)` instead of `&val` as the argument.
+ * After a move, `val` will still exist, but will no longer be valid. The destructors are double-drop-safe, but other functions will still trust that your `val` is valid.
+ *
+ * To check if `val` is still valid, you may use `z_X_check(&val)` (or `z_check(val)` if your compiler supports `_Generic`), which will return `true` if `val` is valid.
+ */
+typedef struct z_hello_t {
+  unsigned int whatami;
+  struct z_id_t pid;
+  const struct z_owned_str_array_t *locators;
+} z_hello_t;
```

Comment:
I am wondering if it makes sense to have a `owned` type within the `z_hello_t`

Reply:
To me, a const pointer to an owned type isn't an issue. However, we do need to have ZP and ZC agree on this type for this field to become public, so I'll make a borrowed type too :)